### PR TITLE
feat: support apiserver query, dbapi close and fix

### PR DESCRIPTION
**Pull request by vagetablechicken**
apiserver query, POST, should set the execute mode in request body.
e.g.
`curl -X POST http://127.0.0.1:9080/dbs/airflow_example -d'{"sql":"select 1", "mode":"offsync"}'`

apiserver get won't return ResultSet now, todo.

- [x] use ExecuteSQL to handle all sql

### CODE REVIEW

**Review by aceforeverd — APPROVED** [review]

File: src/apiserver/api_server_test.cc [hunk 1]
```diff
@@ -14,18 +14,19 @@
...
 #include "apiserver/api_server_impl.h"
 #include "brpc/channel.h"
-#include "memory"
 #include "brpc/restful.h"
 #include "brpc/server.h"
 #include "butil/logging.h"
 #include "gflags/gflags.h"
 #include "gtest/gtest.h"
 #include "json2pb/rapidjson.h"
+#include "memory"
```
Comment (dl239) [inline src/apiserver/api_server_test.cc]: `#include <memory>` ?

File: src/apiserver/api_server_impl.cc [hunk 2]
```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
...
+
+    std::string ToString() const {
+        std::stringstream ss;
+        ss << (is_online ? "online" : "offline") << "-" << (is_sync ? "sync" : "async") << "-" << job_timeout << "ms";
+        return ss.str();
+    }
+};
+
+std::map<std::string, ExecContext> mode_map{
+    {"offsync", {false, true}}, {"offasync", {false, false}}, {"online", {true, false}}};
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc:103]: resolve cpplint error

File: src/apiserver/api_server_impl.cc [hunk 3]
```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
     cntl->response_attachment().append(writer.GetString());
 }
 
+class ExecContext {
+ public:
+    bool is_online = false;
+    bool is_sync = true;
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc:90]: it's better to declare with struct here. refer https://google.github.io/styleguide/cppguide.html#Variable_Names.

File: src/apiserver/api_server_impl.cc [hunk 4]
```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
...
+                       auto mode = boost::to_lower_copy(req.mode);
+                       auto it = mode_map.find(mode);
+                       if (it == mode_map.end()) {
+                           writer << resp.Set("Invalid mode " + mode);
+                           return;
+                       }
+                       ExecContext ctx = it->second;
+
+                       const auto& sql = req.sql;
+                       LOG(INFO) << "post [" << ctx.ToString() << "] query on db [" << db << "], sql: " << sql;
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc]: DLOG

File: src/apiserver/api_server_impl.cc [hunk 5]
```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
...
+                       // TODO(hw): if api server supports standalone, we should check if cluster mode here
+
+                       hybridse::sdk::Status status;
+                       // TODO(hw): if sql is not a query, it may be a ddl, we use ExecuteSQL to execute it before we
+                       //  supports ddl http api. It's useful for api server tests(We can create table when we only
+                       //  connect to the api server).
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc:139]: support executing ddl or insert? #2002

Comment (dl239) [reply src/apiserver/api_server_impl.cc:138]: how to get the running status  if execute an offline async job

File: src/apiserver/api_server_impl.cc [hunk 6]
```diff
@@ -83,6 +83,67 @@ void APIServerImpl::Process(google::protobuf::RpcController* cntl_base, const Ht
...
+                       sql_router_->ExecuteSQL(db, sql, ctx.is_online, ctx.is_sync, ctx.job_timeout, &status);
```
Comment (dl239) [inline src/apiserver/api_server_impl.cc:140]: how to get the status if execute an offline async job

File: src/sdk/sql_cluster_router.h [hunk 7]
```diff
@@ -260,27 +264,21 @@ class SQLClusterRouter : public SQLRouter {
...
-                                                          std::string& output);  // NOLINT
+    std::shared_ptr<hybridse::sdk::ResultSet> ExecuteOfflineQuery(const std::string& db, const std::string& sql,
+                                                                  bool is_sync_job, int job_timeout,
+                                                                  ::hybridse::sdk::Status* status) override;
 
     ::openmldb::base::Status ImportOnlineData(const std::string& sql, const std::map<std::string, std::string>& config,
-                                              const std::string& default_db, bool sync_job,
-                                              ::openmldb::taskmanager::JobInfo& job_info) override;
+                                              const std::string& default_db, bool sync_job, int job_timeout,
+                                              ::openmldb::taskmanager::JobInfo& job_info);
```
Comment (dl239) [inline src/sdk/sql_cluster_router.h]: resolve `cpplint` error

File: src/sdk/sql_cluster_router.cc [hunk 8]
```diff
@@ -2908,14 +2935,14 @@ hybridse::sdk::Status SQLClusterRouter::HandleCreateFunction(const hybridse::nod
     return {};
 }
 
-hybridse::sdk::Status SQLClusterRouter::HandleDeploy(const hybridse::node::DeployPlanNode* deploy_node) {
+hybridse::sdk::Status SQLClusterRouter::HandleDeploy(const std::string& db, const hybridse::node::DeployPlanNode* deploy_node) {
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc]: `cpplint`

File: src/apiserver/api_server_impl.cc (hunk 3 above)
Comment (vagetablechicken) [reply src/apiserver/api_server_impl.cc:90]: done

File: src/apiserver/api_server_impl.cc (hunk 4 above)
Comment (vagetablechicken) [reply src/apiserver/api_server_impl.cc]: VLOG is better, avoid rebuild in debug mode.

File: src/apiserver/api_server_impl.cc (hunk 5 above)
Comment (vagetablechicken) [reply src/apiserver/api_server_impl.cc:139]: Yep. Maybe issue 2002 can be 'write the unit tests for all sql types'? sql parameterized and getting result are two issues 2127 2128

File: src/apiserver/api_server_impl.cc (hunk 6 above)
Comment (vagetablechicken) [reply src/apiserver/api_server_impl.cc:140]: Result check is issue 2127, this pr haven't supported it.

File: src/apiserver/api_server_test.cc (hunk 1 above)
Comment (vagetablechicken) [reply src/apiserver/api_server_test.cc]: old bug, fixed

File: src/sdk/sql_cluster_router.h (hunk 7 above)
Comment (vagetablechicken) [reply src/sdk/sql_cluster_router.h]: We should convert all `::openmldb::taskmanager::JobInfo& job_info` to pointer, but too much changes here. Let's create an issue for it. I'll use `NOLINT` here.

**Review by dl239 — APPROVED** [review]

**Pull request merged**